src/tests/components/registerUser.test.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import axios from 'axios';
+import MockAdapter from 'axios-mock-adapter';
+import configureMockStore from 'redux-mock-store';
+import { RegisterUser } from '../../components/register/RegisterUser';
+
+describe('RegisterUser component', () => {
+  const mockStore = configureMockStore();
+  let store; 
+  let wrapper;
+  const fetchUsers = jest.fn();
+  const createSpy = (toSpy) => jest.spyOn(wrapper.instance(), toSpy);
+
+  const getEvent = (name = '', value = '') => ({
+    preventDefault: jest.fn(),
+    target: {
+      name,
+      value,
+    },
+  });
+
+  beforeEach(() => {
+    store = mockStore({});
+    wrapper = shallow(<RegisterUser fetchUsers={fetchUsers} />);
+  });
+
+  it('should render correctly', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should call handleSubmit on form submission', () => {
+    const spy = createSpy('handleSubmit');
+    wrapper.instance().forceUpdate();
+    wrapper.find('form').simulate('submit', getEvent());
+    expect(spy).toBeCalled();
+  });
+
+  it('should call handleChange on form entry', () => {
+    const spy = createSpy('handleChange');
+    wrapper.instance().forceUpdate();
+    wrapper.find('input[name="email"]').simulate('change', { target: { value: 'some value' } });
+    expect(spy).toBeCalled();
+  });
+
+  describe('handleSubmit', () => {
+    it('should call fetchUsers when called', () => {
+      wrapper.find('#form').simulate('submit', { preventDefault() {} });
+      expect(fetchUsers).toBeCalled();
+    });
+  });
+
+  it('should pass user data form submission', () => {
+    wrapper.find('#email').simulate('change', { target: { name: 'email', value: 'user100@gmail.com' } });
+    wrapper.find('#password').simulate('change', { target: { name: 'password', value: 'Abc12345' } });
+    wrapper.find('#username').simulate('change', { target: { name: 'username', value: 'user100' } });
+    wrapper.find('#form').simulate('submit', { preventDefault() {} });
+    expect(fetchUsers.mock.calls[2][0]).toEqual({ user: { email: 'user100@gmail.com', password: 'Abc12345', username: 'user100' } });
+  });
+
+  it('should redirect on successful login', () => {
```

Comment:
What is this test doing?

Reply:
I had put it there to test whether there is redirection on successful user signup, the test has now been added.